lib/server.js

```diff
@@ -789,33 +772,46 @@ Server.prototype._handle = function _handle(req, res) {
  */
 Server.prototype._route = function _route(req, res, name, cb) {
     var self = this;
+    // helper function to, when on router error, emit error events and then
+    // flush the err.
+    var errResponse = function errResponse(err) {
+        return self._emitErrorEvents(req, res, null, err, function () {
+            res.send(err);
+            return self._finishReqResCycle(req, res, null, err);
+        });
+    };
 
     if (typeof (name) === 'function') {
         cb = name;
         name = null;
 
-        this.router.find(req, res, function onRoute(err, route, ctx) {
+        return this.router.find(req, res, function onRoute(err, route, ctx) {
             var r = route ? route.name : null;
 
             if (err) {
+                // TODO: if its a 404 for OPTION method (likely a CORS
+                // preflight), return OK. This should move into userland.
```

Comment:
I updated the TODO in the `optionsError()` method to reflect what's going on here. This behavior isn't actually new - just moved it out of the `optionsError()` method. Here's the comment I added:

```
/** 
 * returns true if the router generated a 404 for an options request.
 * TODO: this is relevant for CORS only. Should move this out eventually to a
 * userland middleware? This also seems a little like overreach, as there is no
 * option to opt out of this behavior today.
```